### PR TITLE
KAFKA-3835: Streams is creating two ProducerRecords for each send via RecordCollector

### PR DESCRIPTION
**Pull request by jeyhunkarimov**

### CODE REVIEW

**Review by mjsax — APPROVED** [review]
LGTM

**Comment by mjsax** [comment]
\cc @guozhangwang

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java [hunk 1]
```diff
@@ -26,7 +26,13 @@
 public interface RecordCollector {
     <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer);
 
-    <K, V> void send(ProducerRecord<K, V> record, Serializer<K> keySerializer, Serializer<V> valueSerializer,
+    <K, V> void send(final String topic,
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java:34]: I think the first `send()` call can also be in refactored to not have `ProducerRecord` as its parameter: it is only used in the `StoreChangeLogger` which is also an internal class.

**Review by guozhangwang — COMMENTED** [review]
One comment, otherwise LGTM.

**Comment by jeyhunkarimov** [comment]
@mjsax @guozhangwang, I have changed another `send` method

**Comment by guozhangwang** [comment]
Merged to trunk and 0.10.2, thanks @jeyhunkarimov !

**Pull request closed without merging**